.github/workflows/sync.yml

```
@@ -0,0 +1,57 @@
+name: Kernel Sync
+
+on:
+  schedule:
+    - cron: "0 8 * * *"
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: kernel-build-lol
+    container:
+      image: rockylinux:9
+    strategy:
+      fail-fast: false
+      matrix:
+         include:
+           - branch: kernel-mainline
+             remote: git://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git
+             remote_branch: master
+           - branch: stable_6.12.y
+             remote: git://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git
+             remote_branch: linux-6.12.y
+           - branch: centos9
+             remote: https://gitlab.com/redhat/centos-stream/src/kernel/centos-stream-9
+             remote_branch: main
+
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      BRANCH: ${{ matrix.branch }}
+      REMOTE: ${{ matrix.remote }}
+      REMOTE_BRANCH: ${{ matrix.remote_branch }}
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      - name: Install git
+        run: |
+          dnf install -y git
+
+      - name: Sync kernel source
+        run: |
+          set -e
+
+          git clone --branch ${BRANCH} "https://oauth2:$GITHUB_TOKEN@github.com/ctrliq/kernel-src-tree" kernel
+
+          cd kernel
+
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+
+          git remote add upstream ${REMOTE}
+          git fetch upstream --tags --quiet
+          git rebase upstream/${REMOTE_BRANCH}
```

```suggestion
          if ! git rebase "upstream/${REMOTE_BRANCH}"; then
            echo "Error: Rebase failed on branch 'upstream/${REMOTE_BRANCH}', likely due to conflicts"
            git rebase --abort
            exit 1
          fi
```

Added some error handling, rebasing makes me nervous sometimes

---

We have 'set -e' so the script is already going to stop immediately if 'git rebase' fails, right?

---

I usually prefer doing the 
```
RET=$()
if RET -ne 0 
```

But this is a VERY C thing todo

---

I don't tend to use `set -e` so I didn't even think about it.

This is fine by me